torchvision/io/video_reader.py

```diff
@@ -82,30 +105,34 @@ class VideoReader:
             Default value (0) enables multithreading with codec-dependent heuristic. The performance
             will depend on the version of FFMPEG codecs supported.
 
-        device (str, optional): Device to be used for decoding. Defaults to ``"cpu"``.
-            To use GPU decoding, pass ``device="cuda"``.
-
     """
 
-    def __init__(self, path: str, stream: str = "video", num_threads: int = 0, device: str = "cpu") -> None:
+    def __init__(self, path: str, stream: str = "video", num_threads: int = 0) -> None:
         _log_api_usage_once(self)
-        self.is_cuda = False
-        device = torch.device(device)
-        if device.type == "cuda":
-            if not _HAS_GPU_VIDEO_DECODER:
-                raise RuntimeError("Not compiled with GPU decoder support.")
-            self.is_cuda = True
+        from .. import get_video_backend
+
+        self.backend = get_video_backend()
+        print("Initiated the backend", self.backend)
```

Comment:
if we really need this message I would add something like `f"VideoReader initiated with {self.backend} backend"`

Reply:
We don't, it was for debugging purposes only. Good catch